docs/adr/0003-use-docker-volumes-for-improved-macos-performance.md

```diff
@@ -0,0 +1,102 @@
+to a performance issue within GOV.UK Docker.
+
+Sassc is known to [significantly improve the compilation time][sassc-speed] of
+Sass compared to Ruby Sass. When we updated [Content Publisher][] to Sassc
+found that the [application.scss][] file compiled in 20% of the time (~20s to
+4s) when running natively on macOS. However when this was run on GOV.UK Docker
+the performance was significantly less impressive, compiling in 62% of the
+time (~31s to ~19s).
+
+We discovered that the source of this performance issue was the
+[performance overhead of reading][docker-mac-read] from the
+[shared mount][govuk-mount] of application files on macOS. We considered a number
+of options of how this could be improved:
+
+1. Use [Docker volumes][volume] to store non-application code (such as caches or
+   dependencies) to benefit from faster I/O access.
+2. [Configure GOV.UK Docker to use NFS][nfs-docker] for improved I/O access to
+   shared mounts.
+3. Use [docker-sync.io](http://docker-sync.io/) as a means for improved I/O
+   access to shared mounts.
+
+[asset pipeline]: https://guides.rubyonrails.org/asset_pipeline.html
+[504-fix]: https://github.com/alphagov/govuk-docker/pull/310
+[govuk-frontend-performance]: https://github.com/alphagov/govuk-frontend/issues/1671
+[sassc-speed]: https://www.solitr.com/blog/2014/01/css-preprocessor-benchmark/
+[Content Publisher]: https://github.com/alphagov/content-publisher
+[application.scss]: https://github.com/alphagov/content-publisher/blob/2f91d43bcd1eb1f2d30f9d1ff9e556cd4d0de2a6/app/assets/stylesheets/application.scss
+[docker-mac-read]: https://github.com/docker/for-mac/issues/77
+[govuk-mount]: https://github.com/alphagov/govuk-docker/blob/cb124a3a1d3353e777d4f777d77f03f93415d415/projects/content-publisher/docker-compose.yml#L11
+[volume]: https://docs.docker.com/storage/volumes/
+[nfs-docker]: https://www.jeffgeerling.com/blog/2020/revisiting-docker-macs-performance-nfs-volumes
+
+## Decision
+
+We decided to use Docker volumes for non-application code.
+
+When trying this on Content Publisher with the `tmp` directory and
+`node_modules` directory as Docker volumes we saw a significant improvement
+in Sass compilation time on macOS:
+
+| Environment                                           | Ruby Sass | Sassc | Decrease |
+|-------------------------------------------------------|-----------|-------|----------|
+| Native macOS                                          | ~20s      | ~4s   | 80%      |
+| GOV.UK Docker (with shared `~/govuk` mount)           | ~31s      | ~19s  | 38%      |
+| GOV.UK Docker (with `tmp` and `node_modules` volumes) | ~16s      | ~4s   | 75%      |
+
+We saw less impressive improvements by switching the shared mount to use NFS.
+This performed approximately 3 times slower than the Docker volume approach.
+This approach was also not desirable as it required configuration on the host
+machine to create the NFS directory which would complicate installation and
```

Comment:
This is quite a detailed explanation relative to the intro in the previous section. I've suggested we add a little more to explain it there, to try and match the depth here.